src/pages/reactivistas/index.tsx

```diff
@@ -15,121 +13,166 @@ const ReactGroupPage: React.FC = () => {
-                name="study-material"
-                type="text"
-                placeholder="Ingresa el material de estudio"
-                required
-                ref={register({ required: true })}
-              />
-            </div>
-            <div className="mb-4">
-              <label className="block mb-2 text-sm font-bold">
-                Estilo de reuniones
-              </label>
-              <textarea
-                rows={5}
-                className="w-full px-3 py-2 text-sm leading-tight text-gray-700 border rounded appearance-none focus:outline-none focus:shadow-outline"
-                name="meeting-type"
-                placeholder="Ingresa la dinamica que se va a utilizar en su grupo de estudio"
-                required
-                ref={register({ required: true })}
-              ></textarea>
-            </div>
-            <div className="mb-4">
-              <label className="block mb-2 text-sm font-bold">
-                Plan de estudio
-              </label>
-              <textarea
-                rows={5}
-                className="w-full px-3 py-2 text-sm leading-tight text-gray-700 border rounded appearance-none focus:outline-none focus:shadow-outline"
-                name="plan"
-                placeholder="Ingresa como va a ser el plan de estudio"
-                required
-                ref={register({ required: true })}
-              ></textarea>
-            </div>
-            <div className="mb-4">
-              <label className="block mb-2 text-sm font-bold">
-                Fecha tentativa de inicio
-              </label>
-              <input
-                className="w-full px-3 py-2 text-sm leading-tight text-gray-700 border rounded appearance-none focus:outline-none focus:shadow-outline"
-                name="date"
-                type="date"
-                required
-                ref={register({ required: true })}
-              />
-            </div>
-            <div className="pt-8">
-              <div className="flex justify-end">
-                <button
-                  type="submit"
-                  className="inline-flex justify-center px-4 py-2 ml-3 text-sm font-medium text-white border border-transparent rounded-md shadow-sm bg-primary hover:bg-primarydark focus:outline-none focus:ring-2 focus:ring-offset-2 focus:ring-indigo-500"
-                >
-                  Enviar propuesta de grupo
-                </button>
-              </div>
-            </div>
-          </form>
+          <ReactGroupForm />
         </div>
       </div>
     </Layout>
   );
 };
 
+const ReactGroupForm = () => {
+  const { register, handleSubmit } = useForm();
+  const [message, setMessage] = useState<string>();
+  const [error, setError] = useState<boolean>(false);
+
+  const onSubmit = async (data: ReactGroup) => {
```

Comment:
Can we put this logic on a `/pages/api` page, so we dont have to expose Sanity token to the browser? Then we would just call that function (`/api/theFunction`) from here.